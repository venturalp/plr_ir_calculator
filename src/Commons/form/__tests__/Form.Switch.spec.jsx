import React from 'react'
import { render } from '@testing-library/react'
import { ThemeWrapper } from 'Commons/tests/Tests.Helpers'
import { CustomSwitch } from '../Form.Switch'

const renderSwitch = props =>
  render(
    <ThemeWrapper>
      <CustomSwitch {...props} />
    </ThemeWrapper>,
  )

describe('test form switch component', () => {
  it('render with default colors', () => {
    const { asFragment } = renderSwitch()

    expect(asFragment()).toMatchSnapshot()
  })
  it('render with custom color', () => {
    const { asFragment } = renderSwitch({ fill: '#ff0000' })

    expect(asFragment()).toMatchSnapshot()
  })
})
