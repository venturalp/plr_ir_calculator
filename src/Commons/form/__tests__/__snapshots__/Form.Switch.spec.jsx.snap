// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test form switch component render with custom color 1`] = `
<DocumentFragment>
  <span
    class="MuiSwitch-root makeStyles-switch-7 makeStyles-switch-8"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiSwitch-switchBase MuiSwitch-colorSecondary"
      fill="#ff0000"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-12 MuiSwitch-input"
          type="checkbox"
          value=""
        />
        <span
          class="MuiSwitch-thumb"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>
</DocumentFragment>
`;

exports[`test form switch component render with default colors 1`] = `
<DocumentFragment>
  <span
    class="MuiSwitch-root makeStyles-switch-1 makeStyles-switch-2"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiSwitch-switchBase MuiSwitch-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-6 MuiSwitch-input"
          type="checkbox"
          value=""
        />
        <span
          class="MuiSwitch-thumb"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>
</DocumentFragment>
`;
